akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala

```diff
@@ -186,6 +186,12 @@ object Source {
       override def toString: String = "() => Iterator"
     })
 
+  def fromJavaStream[T, S <: java.util.stream.BaseStream[T, S]](stream: () ⇒ java.util.stream.BaseStream[T, S]): Source[T, Unit] = {
+
+    import scala.collection.JavaConverters._
+    Source.fromIterator(() ⇒ stream().iterator().asScala)
```

Comment:
this should probably be documented to benefit from an asyncBoundary in certain cases (once you get to writing docs, that is :-) )